hdr/sqlite_modern_cpp.h

```diff
@@ -362,7 +370,7 @@ namespace sqlite {
 		std::shared_ptr<sqlite3> _db;
 
 	public:
-		database(const std::string &db_name, const sqlite_config &config = {}): _db(nullptr) {
+		database(const STR_REF &db_name, const sqlite_config &config = {}): _db(nullptr) {
 			sqlite3* tmp = nullptr;
 			auto ret = sqlite3_open_v2(db_name.data(), &tmp, static_cast<int>(config.flags), config.zVfs);
```

Comment:
This assumes that `db_name` is null-terminated. The `db_name` ends up as a filenamem, so not null-terminated strings are not supported here.
Maybe we should keep `std::string` here and optionally add a  `const char *` overload?